### PR TITLE
Fix puppi event content and JEC

### PR DESCRIPTION
**Pull request by gpetruc**
- Keep puppi jets (by mistake they were dropped, because of a conflict between the subjet PR and puppi PR)
- Use AK4PFchs L2+L3 corrections on puppi jets (no L1, since existing L1 corrections for PF or PFchs are not appropriate for puppi)
- Properly compute jet area for puppi jets (allow application of L1 corrections in the future when they will be available for these jets)

@arizzi @jshlee @violatingcp

### CODE REVIEW

**Comment by davidlange6** [comment]
@cmsbuild, please test

File: PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py [hunk 1]
```diff
@@ -161,7 +162,7 @@ def miniAOD_customizeCommon(process):
     )
 
     addJetCollection(process, postfix   = "", labelName = 'Puppi', jetSource = cms.InputTag('ak4PFJetsPuppi'),
-                    jetCorrections = ('AK4PF', ['L1FastJet', 'L2Relative', 'L3Absolute'], ''),
+                    jetCorrections = ('AK4PFchs', ['L2Relative', 'L3Absolute'], ''),
                     algo= 'AK', rParam = 0.4, btagDiscriminators = map(lambda x: x.value() ,process.patJets.discriminatorSources)
                     )
```
Comment (ferencek) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: L173 and 174 should be removed. SV TagInfos don't seem to be used anywhere so there is no point in adding them and later dropping them in the process of slimming.

Comment (jshlee) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: Shall we do a different PR for this or add it into this one?

Comment (gpetruc) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: unless the idea was to instead add the equivalent of L105-119 for Puppi jets as well.
@jshlee can you see how much disk space does it take, on the usual 9k ttbar events?

Comment (jshlee) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: @gpetruc, as @ferencek mentioned, SV TagInfos are not saved in slimmedJets and slimmedJetsPuppi (only slimmedJetsAK8), so we don't need to save them in patJets. Similarly, L120-L121 can be removed for patJets too.
I can do a check on the disk space usage of this branch.

Comment (jshlee) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: @gpetruc and @ferencek, I just realised that the SV TagInfos are required by the userData, so we need to keep it while making pat jets.

Comment (ferencek) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: As @gpetruc said, it is needed if the intention is to store some of the SV info as userFloats. If not, then there is no need to add TagInfos. Also note that for slimmedJetsAK8 TopTagInfos are stored for later access to the top tagging information.
